### PR TITLE
feat(suite): show zero balance tokens

### LINKED ISSUE #13338: EVM Tokens/ Allow user to show 0 balance tokens
Prerequisite for #13039

Allow user to show 0 balance tokens to be able to operate with all tokens he has in tx history and if he want to trade it

Rabby also shows it under this button
Screenshot 2024-07-15 at 12 38 55

Design:
image

### PR DESCRIPTION
**Pull request by AdamSchinzel**
## Description

Show zero balance tokens instead of hiding them behind debug mode.

## Related Issue

Resolve #13338

## Screenshots:
Screenshot 2024-07-31 at 9 35 46 AM

### CODE REVIEW

File: packages/suite/src/utils/wallet/tokenUtils.ts [hunk 1]
```diff
@@ -90,16 +90,41 @@ export const getTokens = (
 
         if (searchQuery && !isTokenMatchesSearch(token, query)) return;
 
+        const hasBalance = token.balance !== '0';
```
Comment (tomasklim) [inline packages/suite/src/utils/wallet/tokenUtils.ts]: Please use BigNumber.gte(0), just to be sure

File: packages/suite/src/utils/wallet/tokenUtils.ts [hunk 2]
```diff
@@ -90,16 +90,41 @@ export const getTokens = (
...
+
         if (isShown) {
```
Comment (tomasklim) [inline packages/suite/src/utils/wallet/tokenUtils.ts]: how about something like this, but not strong opinion 

```
const hasBalance = token.balance !== '0';
const pushToArray = (arrayWithBalance, arrayWithoutBalance) => {
    if (hasBalance) {
        arrayWithBalance.push(token);
    } else {
        arrayWithoutBalance.push(token);
    }
};

if (isShown) {
    pushToArray(shownWithBalance, shownWithoutBalance);
} else if (hasCoinDefinitions && !isKnown) {
    pushToArray(unverifiedWithBalance, unverifiedWithoutBalance);
} else if (isHidden) {
    pushToArray(hiddenWithBalance, hiddenWithoutBalance);
} else {
    pushToArray(shownWithBalance, shownWithoutBalance);
}
```

**Review by tomasklim — COMMENTED** [review]
Otherwise looks good. Please create test for `getTokens` it's getting really complicated

File: packages/suite/src/utils/wallet/tokenUtils.ts (hunk 2 above)
Comment (AdamSchinzel) [reply packages/suite/src/utils/wallet/tokenUtils.ts]: Looks cleaner 👍

**Review by tomasklim — APPROVED** [review]

**Pull request merged**